bitcoin/src/p2p/mod.rs

```diff
@@ -243,31 +243,38 @@ impl FromStr for Magic {
     }
 }
 
-impl From<Network> for Magic {
-    fn from(network: Network) -> Magic {
-        match network {
-            // Note: new network entries must explicitly be matched in `try_from` below.
-            Network::Bitcoin => Magic::BITCOIN,
-            Network::Testnet => Magic::TESTNET,
-            Network::Signet => Magic::SIGNET,
-            Network::Regtest => Magic::REGTEST,
+macro_rules! generate_network_magic_conversion {
+    ($(Network::$network:ident => Magic::$magic:ident,)*) => {
```

Comment:
Writing `Network::$network:ident` instead of `$network:path` is a bit confusing but I think it's fine. The usage below makes it obvious how to add a new one and we probably won't need to change the macro.

Reply:
yes, should be obvious, happy to change it though to any other syntax